File: solutions/c/scrabble-score/1/scrabble_score.c
#include "scrabble_score.h"

static unsigned int letter_score(char let) {
    switch (let) {
        case 'q': case 'z': return 10;
        case 'j': case 'x': return 8;
        case 'k': return 5;
        case 'f': case 'h': case 'v': case 'w': case 'y': return 4;
        case 'b': case 'c': case 'm': case 'p': return 3;
        case 'd': case 'g': return 2;
        default: return 1;
    }
}

unsigned int score(const char *word) {
    unsigned int score = 0;
    for (; *word; word++) {
        score += letter_score(*word <= 'Z' ? *word + 32 : *word);
    }
    return score;
}

File: solutions/python/scrabble-score/1/scrabble_score.py
SCORES = {
    "Q": 10,
    "Z": 10,
    "J": 8,
    "X": 8,
    "K": 5,
    "F": 4,
    "H": 4,
    "V": 4,
    "W": 4,
    "Y": 4,
    "B": 3,
    "C": 3,
    "M": 3,
    "P": 3,
    "D": 2,
    "G": 2,
}


def score(word):
    return sum(SCORES.get(c.upper(), 1) for c in word if c.isalpha())
